### ISSUE TITLE
PadArray() doesn't support pads with no number

### ISSUE THREAD

**Issue by evanshultz** [pointhi/kicad-footprint-generator#323]
Pad() allows pads with an empty string for the `number` parameter which gives unnumbered pads. This is used for NPTH pads. However, PadArray() doesn't support this because the initialPin variable must be an integer. It would be nice to support NPTHs with no number as an array.

**Comment by poeschlr** [comment]
Yes and bga style numbering or any non int numbering scheme would also be nice. (would either need the pin number to be made as a class that defines addition or there would also need to be a way to define different pin number increment functions.)
